Library/Homebrew/cmd/update.rb

```diff
@@ -216,6 +216,20 @@ def pull!(options = {})
       @initial_branch = ""
     end
 
+    # used for testing purpose, for example for testing formula migrations when
+    # some formula gets renamed on your branch and you test migrations by
+    # running 'brew update --from-branch"
```

Comment:
I'd rephrase the comment as follows to make it a bit clearer (I'm not a native speaker): “Used for testing purposes, e.g., for testing formula migrations in the currently checked-out branch. To test, run `brew update --from-branch`.”